### PR TITLE
automatically add traces / metrics paths

### LINKED ISSUE #773: Add signal path automatically for OTLP HTTP exporter
As per [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#example-1), OTLP http exporter should be able to automatically add a path based on signal.

### PR DESCRIPTION
**Pull request by GaryPWhite**
Resolves #773

### CODE REVIEW

**Comment by GaryPWhite** [comment]
noting that we could probably add logs exporter -- I noticed there was documentation for log signals, but the logs.rs exporter doesn't exist :)

**Comment by TommyCpp** [comment]
> noting that we could probably add logs exporter -- I noticed there was documentation for log signals, but the logs.rs exporter doesn't exist :)

Ahh yeah once we merged #788 we can implement a logs exporter

File: opentelemetry-otlp/src/exporter/mod.rs [hunk 1]
```diff
@@ -12,20 +12,15 @@ use crate::Protocol;
 use std::str::FromStr;
 use std::time::Duration;
 
-/// Target to which the exporter is going to send spans or metrics, defaults to https://localhost:4317.
```
Comment (TommyCpp) [inline opentelemetry-otlp/src/exporter/mod.rs:15]: :+1: Like this doc change

File: opentelemetry-otlp/src/span.rs [hunk 2]
```diff
@@ -63,6 +64,11 @@ use opentelemetry::{
 
 use async_trait::async_trait;
 
+/// Target to which the exporter is going to send spans, defaults to https://localhost:4317.
```
Comment (TommyCpp) [inline opentelemetry-otlp/src/span.rs:67]:
```suggestion
/// Target to which the exporter is going to send spans, defaults to https://localhost:4317/v1/traces.
```

**Review by TommyCpp — COMMENTED** [review]
I think it generally looks good. Thanks!

One ask I have is maybe we can add some document to explain the relationship between `OTEL_EXPORTER_OTLP_END` and `OTEL_EXPORTER_OTLP_{signal}_ENDPOINT` like https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#endpoint-urls-for-otlphttp.

And reference it from env var constants. It may be easier for user to understand what to choose

File: opentelemetry-otlp/src/exporter/mod.rs (hunk 1 above)
Comment (GaryPWhite) [reply opentelemetry-otlp/src/exporter/mod.rs:15]: thanks much!

**Comment by GaryPWhite** [comment]
> > noting that we could probably add logs exporter -- I noticed there was documentation for log signals, but the logs.rs exporter doesn't exist :)
> 
> Ahh yeah once we merged #788 we can implement a logs exporter

I can make an issue or continue from #773 when it is merged! feel free to assign me or let me know here!

**Review by TommyCpp — APPROVED** [review]
It seems log may need sometime to land. I think we can close #773 and create a new one for add log path and related envionment variables. 

The change looks nice 👍

**Comment by jtescher** [comment]
Just waiting on tests/lints now?

**Comment by GaryPWhite** [comment]
sorry about that -- doesn't run without maintainers clicking it so the feedback loop is long :)

**Comment by TommyCpp** [comment]
> sorry about that -- doesn't run without maintainers clicking it so the feedback loop is long :)

It only applies to the first PR(to prevent bad actors from injecting crypto code in PR to exploit CI pipeline). Should be better once it's merged

**Comment by TommyCpp** [comment]
Running `./precommit.sh` can help find the lint issue

**Comment by GaryPWhite** [comment]
alright let me run that and be back in a sec

**Comment by TommyCpp** [comment]
Emm looks like the EasyCLA stuck again 😞

**Comment by GaryPWhite** [comment]
no worries. I'll find something to fix i'm sure with the script and push a new one up

**Pull request merged**